src/crimson/osd/ops_executer.cc

```diff
@@ -303,45 +303,41 @@ OpsExecuter::watch_errorator::future<> OpsExecuter::do_op_notify_ack(
 {
   struct notifyack_ctx_t {
     const entity_name_t entity;
-    std::optional<uint64_t> watch_cookie;
+    uint64_t watch_cookie;
     uint64_t notify_id;
     ceph::bufferlist reply_bl;
 
     notifyack_ctx_t(const MOSDOp& msg) : entity(msg.get_reqid().name) {
     }
   };
   return with_effect_on_obc(notifyack_ctx_t{ get_message() },
-    [&] (auto& ctx) {
+    [&] (auto& ctx) -> watch_errorator::future<> {
       try {
         auto bp = osd_op.indata.cbegin();
-        uint64_t wc = 0;
         ceph::decode(ctx.notify_id, bp);
-        ceph::decode(wc, bp);
-        ctx.watch_cookie = wc;
+        ceph::decode(ctx.watch_cookie, bp);
         if (!bp.end()) {
           ceph::decode(ctx.reply_bl, bp);
         }
       } catch (const buffer::error&) {
-        // op.watch.cookie is actually the notify_id for historical reasons
-        ctx.notify_id = osd_op.op.watch.cookie;
```

Comment:
@athanatos: does dropping this legacy look right to you? `librbd` from current `master` seems to not complain.

Reply:
Yeah, we don't need to support legacy encodings.